### ISSUE TITLE
Handle set-cookie when length exceeds maximum length

### ISSUE THREAD

**Issue by RobertoPrevato** [Neoteroi/BlackSheep#96]
Example log from Opera:

```
Set-Cookie header is ignored in response from url: http://localhost:44777/.... Cookie length should be less than or equal to 4096 characters.
```

**Comment by RobertoPrevato** [comment]
Either throw exception or implement chunking and aggregating logic.

**Comment by myusko** [comment]
@RobertoPrevato based on https://tools.ietf.org/html/rfc6265#section-6.1

Probably we should throw the exception, and don't consider chunking, what do you think?

**Comment by RobertoPrevato** [comment]
@myusko I am a bit undecided. Throwing exception makes sense and is a perfectly legitimate thing to do.
On the other hand, it could be a better user experience for programmers if the web framework emitted a warning and automatically handled the situation, creating several cookies. I recall that I saw this practice somewhere in the past, like some server creating `COOKIE_NAME_1`, `COOKIE_NAME_2` and merging them transparently on the server.

Standards are one side of the medal, but software often do other things, depending on notions of user friendliness or what makes sense for developers. For example, db implementations vary greatly about some SQL standards (I think about MERGE commands in SQL Server, PostgreSQL, MySQL, and also different methods to paginate results).

**Comment by RobertoPrevato** [comment]
Hi @myusko,
after thinking a bit, I decided to follow your recommendation and throw exception for this case. In the future I will add two helper methods to handle bigger cookies, but they will need to be used explicitly.

This exception will be included in `1.0.4` (publishing soon).

**Comment by RobertoPrevato** [comment]
Fixed at #125

**Comment by myusko** [comment]
@RobertoPrevato I love it! let's make BlackSheep better 🚀
